### PR TITLE
MNT: Do not use specreduce v1.1.0, add specreduce to devdeps job

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to workaround #1585 so it does not block unrelated PRs.

Also add specreduce to devdeps job so we know any future breakage before it is too late (hopefully).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by kecnry — APPROVED** [review]
Won't be necessary once #1555 is merge (which pins to 1.1.0 and accounts for the API changes), but we can merge this in the meantime if needed for CI in other PRs.

**Comment by pllim** [comment]
I'll let you decide. If #1555 is merged first, then I can modify the PR to only add devdeps part.

**Pull request merged**
